queries/atm.py

```diff
@@ -0,0 +1,571 @@
+            )
+    return atm_data
+
+
+def _group_closest_based_on_address(
+    atm_data: List[Dict[str, Any]]
+) -> List[Dict[str, Any]]:
+    """
+    Group closest ATMs based on address,
+    returns all atms with the same address or
+    an empty list if no atms are found.
+    """
+    if len(atm_data) == 0:
+        return []
+    atms: List[Dict[str, Any]] = []
+    atms.append(atm_data[0])
+    for idx, atm in enumerate(atm_data[:-1]):
+        if atm["address"]["street"] == atm_data[idx + 1]["address"]["street"]:
+            atms.append(atm_data[idx + 1])
+        else:
+            break
+    return atms
+
+
+def _closest_atm(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _closest_atm_deposit(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location that accepts deposits."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    filtered_atms: List[Dict[str, Any]] = []
+    for atm in atms:
+        services = atm.get("services", {})
+        if services.get("deposit", False) is True:
+            filtered_atms.append(atm)
+    atms = filtered_atms
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _closest_atm_foreign_exchange(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location that accepts foreign exchange."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    filtered_atms: List[Dict[str, Any]] = []
+    for atm in atms:
+        services = atm.get("services", {})
+        if services.get("foreign_exchange", {}).get("active", False) is True:
+            filtered_atms.append(atm)
+    atms = filtered_atms
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
+
+
+def _closest_atm_coinmachine(loc: LatLonTuple) -> List[Dict[str, Any]]:
+    """Find ATM closest to the given location that has a coinmachine."""
+    atms: Optional[List[Dict[str, Any]]] = _atms_with_distance(loc)
+    if not atms:
+        return []
+
+    filtered_atms: List[Dict[str, Any]] = []
+    for atm in atms:
+        services = atm.get("services", {})
+        if services.get("coinmachine", False) is True:
+            filtered_atms.append(atm)
+    atms = filtered_atms
+
+    # Sort by distance
+    dist_sorted = sorted(atms, key=lambda s: s["distance"])
+    return _group_closest_based_on_address(dist_sorted)
```

Comment:
These three lines are repeated in lines 211-213 & 247-249, couldn't we just include the `sorted(atms, ...)` call as the first step in `_group_closest_based_on_address()`?

Reply:
Fixed, the sort call is now the first step in `_group_closest_based_on_address()`.